src/modal/ShowDetailsModal.jsx

```diff
@@ -0,0 +1,91 @@
+/* eslint-disable react/prop-types */
+import { useContext, useEffect } from "react";
+import styled, { css } from "styled-components";
+import { Backdrop, CircularProgress } from "@mui/material";
+import { Theme } from "../context";
+import { useFetch } from "../hooks/useFetch";
+import { useGetCharDetails } from "../hooks/useGetCharDetails";
+import { Button, Image, Wrapper } from "../components";
+
+import FilmsDetails from "./FilmsDetails";
+
+const DetailsInfo = styled.div`
+  margin: 10% auto;
+  display: flex;
+  padding: 10px;
+  text-align: center;
+  align-items: stretch
+  justify-content: center;
+  border-radius: 32px;
+  width: auto;
+  max-width: 1000px;
+  height: auto;
+  max-height: 200px;
+   ${(style) => css`
+     background-color: ${style.backgroundColor};
+     color: ${style.color};
+     border-color: ${style.borderColor};
+     box-shadow: ${style.boxShadow};
+   `}
+`;
+
+const ShowDetailsModal = ({ id, onClick, open }) => {
+  const { characters, loading, errors } = useFetch("1", id);
+  const { themeStyles } = useContext(Theme);
+  const { charDetails } = useGetCharDetails(characters);
+
+  useEffect(() => {
```

Comment:
This logic could be eliminated if the styles were applied to the highest div instead of the body element.